pkg/registration/spoke/spokeagent.go

```diff
@@ -418,6 +459,54 @@ func (o *SpokeAgentConfig) RunSpokeAgentWithSpokeInformers(ctx context.Context,
 		)
 	}
 
+	var hubAcceptController, hubTimeoutController factory.Controller
+	hubKubeConfigSecretAutoGenCtx, cancelHubKubeConfgiSecretAutoGen := context.WithCancel(ctx)
+	defer cancelHubKubeConfgiSecretAutoGen()
+	if features.SpokeMutableFeatureGate.Enabled(ocmfeature.MultipleHubs) {
+		hubAcceptController = registration.NewHubAcceptController(
+			o.agentOptions.SpokeClusterName,
+			hubClusterInformerFactory.Cluster().V1().ManagedClusters(),
+			func(ctx context.Context) error {
+				logger.Info("Failed to connect to hub because of hubAcceptClient set to false, restart agent to reselect a new bootstrap kubeconfig")
+				cancelHubKubeConfgiSecretAutoGen()
+				// remove hub kubeconfig secret
+				if err := managementKubeClient.CoreV1().Secrets(o.agentOptions.ComponentNamespace).Delete(ctx,
```

Comment:
do we really need to remove the secret? it will be restarted with the changed bootstrap which makes the existing hubkubeconfig invalid, right?

Reply:
No, as I mentioned in https://github.com/open-cluster-management-io/ocm/pull/443#discussion_r1609224184, in MultipleHubs enabled cause, we don't compare hub-kubeconfig and bootstrap-kubeconfig.

Reply:
Remove.